src/main/webapp/app/exercises/programming/manage/programming-exercise.component.ts

```diff
@@ -161,6 +166,9 @@ export class ProgrammingExerciseComponent extends ExerciseComponent implements O
         modalRef.componentInstance.selectedProgrammingExercises = this.selectedProgrammingExercises;
     }
 
+    castToParticipation(participation: SolutionProgrammingExerciseParticipation | TemplateProgrammingExerciseParticipation): Participation {
```

Comment:
guess, we decided not using this kind of casting function during the testing session. other than that, seems nice :)

Reply:
I removed the cast, now it should be better